provision/provision.go

```diff
@@ -0,0 +1,162 @@
+// This package will fetch the recommendation from the recommendation Queue and provision the scale in/out
+// based on command.
+package provision
+
+import (
+	"scaling_manager/cluster"
+	"time"
+)
+
+var counter uint8 = 1
+
+// This struct contains the State of the opensearch scaling manager
+// States can be following:
+//  1. provision
+//  2. provisioning
+//  3. provisioning_completed
+//  4. provisioning_failed
+//  5. provisioned_successfully
```

Comment:
Add one more state open_to_provision such that we can decide if we can provision the recommendation or not.